docs/modules/ROOT/pages/references/parameters.adoc

```diff
@@ -240,70 +255,102 @@ default:: `all`
 type:: list
 default:: `[]`
 
-
-== `database.builtin`
+== `database.provider`
 
 [horizontal]
-type:: bool
-default:: `true`
-
-Use Bitnami Postgres installed by the Keycloak chart by default.
+type:: string
+values:: `builtin`, `external`
+default:: `builtin`
 
 
-== `database.external.secretname`
+== `database.database`
 
 [horizontal]
 type:: string
-default:: `keycloak-db-credentials`
+default:: `keycloak`
 
 
-== `database.external.vendor`
+== `database.username`
 
 [horizontal]
 type:: string
-default:: `postgres`
+default:: `keycloak`
 
 
-== `database.external.host`
+== `database.password`
 
 [horizontal]
 type:: string
-default:: `postgres.example.com`
+default:: Vault reference
 
+A Vault reference pointing to the Vault secret containing the Keycloak admin password.
 
-== `database.external.port`
+It is required to pregenerate the password in Vault.
```

Comment:
This is only true when using the built-in database, right? As far as I understand, the provided password for the external database needs to be stored in vault otherwise.

Reply:
for both providers it's required to store it in Vault. I removed this sentence as it doesn't add much value anyway.

Reply:
I was more confused because "pregenerate" implied that a random password is generated, which is not true when using an external DB as far as I understand.